File: CodeChef/Competitions/August Long/rainbow.py
t= int(input())
nos=[1,2,3,4,5,6,7]

for test in range(t):
	n = int(input())
	l = list(map(int,input().split(' ')))
	count=0
	i=0
	if n >= 7 and n%2==1:
		while i<n:
			if l[i]==l[n-i-1] and l[i] in nos:
				count=count+1
				i=i+1
			else:
				break
		if count==n:
			print("yes")
		else:
			print("no")
		i=0
	else:
		print("no")

File: CodeChef/Competitions/August Long/rainbow.cpp
#include <iostream>
#include <math.h>
#include <stdio.h>
using namespace std;
int main(int argc, char const *argv[])
{
	int t;
	cin>>t;
	for(int test =0;test<t;test++)
	{
		int n;
		int i,j;
		cin>>n;
		int count=0;
		int arr[n];
		for(i=0;i<n;i++)
			cin>>arr[i];
		for(i=0,j=n-1;i<=j;i++,j--)
			if(arr[i]==arr[j] && (arr[i]==1||arr[i]==2||arr[i]==3||arr[i]==4||arr[i]==5||arr[i]==6||arr[i]==7))
				count++;

		int res = ceil((n/2.0));
		//cout<<res;
		if(count==res)
			cout<<"yes"<<endl;
		else
			cout<<"no"<<endl;
	}
	return 0;
}
